src/core/public/chrome/chrome_service.tsx

```diff
@@ -163,6 +163,15 @@ export class ChromeService {
       );
     }
 
+    const settings = injectedMetadata.getLegacyMetadata().uiSettings;
+
+    const navSetting =
+      settings?.user?.pageNavigation?.userValue || settings.defaults.pageNavigation.value;
+
+    const license = (injectedMetadata.getInjectedVars().xpackInitialInfo as {
+      license: { type: string };
+    }).license.type;
+
```

Comment:
don't know enough about this (and alternatives) to have a strong objection, but relying on the legacy metadatas on NP services seems like a code smell. thoughts @joshdover ?